### PR TITLE
feat(v2): add Combobox (dropdown) component

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR adds a combobox component that will be used to create a dropdown field in a future PR. This component is also one of the primitive components that will be added into OGP's design system.

Dropdown field implementation will come in a future PR

Related to #2007 

## Solution

**Breaking Changes** 

- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible  

**Features**:
- New `Combobox` (search + dropdown) component

## Before & After Screenshots
In storybook

## Tests

sorry

## Deploy Notes

**New dependencies**:

- `downshift`: hooks for creating an accessible combobox
- `match-sorter`: hooks for filtering and sorting results
- `react-popper`: for dynamic dropdown placement

### CODE REVIEW

**Comment by karrui** [comment]
Converting to draft, I will revamp this component too

File: frontend/src/components/Combobox/Combobox.tsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+import { ToggleChevron } from './ToggleChevron'
+import { ComboboxItem } from './types'
+import {
+  defaultFilter,
+  itemToIcon,
+  itemToLabelString,
+  itemToValue,
+} from './utils'
+
+export interface ComboboxProps<Item = ComboboxItem, Value = string>
```
Comment (mantariksh) [inline frontend/src/components/Combobox/Combobox.tsx:31]: aren't we being inconsistent about `Value` being a generic type? for instance, despite the fact that `Value` is generic, `itemToValue` always returns a `string`. imo there's no need to overengineer this, we can just let it be `string`

File: frontend/src/components/Combobox/Combobox.tsx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(
+      null,
+    )
+    const { styles: popperStyles, attributes: popperAttributes } = usePopper(
+      referenceElement,
+      popperElement,
+      { placement: 'bottom-start' },
+    )
+
+    const { isDisabled, isInvalid, isReadOnly, id } = useFormControlProps({
```
Comment (mantariksh) [inline frontend/src/components/Combobox/Combobox.tsx:106]: lol https://github.com/chakra-ui/chakra-ui/issues/4513

File: frontend/src/components/Combobox/types.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      /** Description to render below label if provided */
+      description?: string
+      /** Whether item is disabled */
+      disabled?: boolean
+      /** Icon to display in input field when item is selected, if available */
+      icon?: As
+      [key: string]: any
+    }
+  | string
+  | null
```
[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/components/Combobox/types.ts:18]: it leaves the possibility for users of this component to just pass in a `string`. 

`null` on the other hand allows us to reuse the type to set "empty" selected items

**Pull request closed without merging**